File: Android/app/src/main/java/com/quickdev/projectdashboard/models/domain/Project.kt
package com.quickdev.projectdashboard.models.domain

import androidx.room.*
import com.quickdev.projectdashboard.util.converters.DateTimeConverter
import java.time.LocalDateTime

@Entity(
    tableName = "projects",
    indices = [
        Index(value = ["id"], unique = true)
    ]
)
@TypeConverters(value = [DateTimeConverter::class])
data class Project(
    @PrimaryKey(autoGenerate = false)
    val id: Int = 0,
    val name: String,
    val lastEdit: LocalDateTime,
    val teamId: Int,
    val ownerId: Int,

    @Embedded
    val contact: ContactInfo
) {
    /** Variable to be filled in by the repository, not saved in the Room */
    @Ignore
    var owner: Company? = null

    /** Variable to be filled in by the repository, not saved in the Room */
    @Ignore
    var team: Team? = null

    /** Variable to be filled in by the repository, not saved in the Room */
    @Ignore
    var tasks: List<ProjectTask> = listOf()
}

File: Backend/ProjectDashboard/Models/Domain/Project.cs
using System;
using System.Collections.Generic;

namespace ProjectDashboard.Models.Domain {
	public class Project {

		public int Id { get; set; }
		public string Name { get; set; }

		public int TeamId { get; set; }
		public Team Team { get; set; }


		public ICollection<ProjectTask> Tasks { get; set; } = new List<ProjectTask>();

		public DateTimeOffset LastEdit { get; set; } = DateTimeOffset.Now;

		public int OwnerId { get; set; }

		public Company Owner { get; set; }

		public ContactInfo ContactPerson { get; set; }

		public Project() { }

		public Project(string name, Team team, Company owner, ContactInfo contact) {
			this.Name = name;
			this.Team = team;
			this.Owner = owner;
			this.ContactPerson = contact;
		}
	}
}
